core/src/edu/cornell/gdiac/temporary/GameplayController.java

```diff
@@ -546,18 +462,18 @@ public void handleActions(InputController input){
 		//We do not want one hit to count for two notes that are close together.
 		boolean[] hitReg = new boolean[triggers.length];
 		int checkBM = curP == play_phase.NOTES ? activeBM : goalBM;
-		for(Note n : bandMembers[checkBM].getHitNotes()){
+		for(Note n : level.getBandMembers()[checkBM].getHitNotes()){
 			if(n.getNoteType() == Note.NoteType.BEAT){
 				if(triggers[n.getLine()] && !hitReg[n.getLine()]){
-					//Check for all the beats in this line and in the active band member
+					//Check for all the notes in this line and in the active band member
 					//See if any are close enough
-					float dist = Math.abs(hitbarY - n.getY())/n.getHeight();
-					if(dist < 1.5){
+					long dist = Math.abs(currentSample - n.getHitSample());
+					if(dist < 25000){
```

Comment:
just saying rn for every part that has logic with checking to see if it's on beat, this is where calibration comes into play. I really recommend separating this into a separate method so this logic doesn't get repeated or cluttered (like I did with `isOnBeat`. You don't have to worry about this because I will likely be working to refactor it, but like I said, ms or samples.

A lot of this seems unlabeled so it's slightly confusing.

Reply:
it is yeah